hokusai/commands/diff.py

```diff
@@ -28,4 +28,7 @@ def diff():
       return -1
 
   print_green("Comparing %s to %s" % (production_tag, staging_tag))
-  shout("git diff %s %s" % (production_tag, staging_tag), print_output=True)
+  if pr_only:
+    shout("git log --merges --right-only %s..%s" % (production_tag, staging_tag), print_output=True)
```

Comment:
Yay! 👍 💯 🎉  to a PR only option! It may be tricky to get precisely every and only the Github PRs. Some questions I had before when thinking about this:

* Omit merge commits other than the real PR merges. _I think_ this implementation may still include them.
* Handle PRs that were squashed and didn't have a separate PR merge commit.

Maybe we don't need to be perfect here (but we may need to make this option/description clear).

In @jonallured's [deploy_pr](https://github.com/jonallured/deploy_pr) script, he had to do something like [this](https://github.com/jonallured/deploy_pr/blob/cf8e85dca6cd6342f35e20602d233adfe70c3d70/deploy_pr#L42), but there were still missed cases.

Reply:
Also, I feel a clean commit-level diff may be helpful too.

Reply:
I agree that a commit-level diff is helpful... I actually assumed that's what `diff` did. And narrowing to just merged PRs is hard when you consider squashes, different forking/upstream practices, etc.

Really `git diff` and `git log` are _such_ different operations that I expect they'll want to support different parameters, too. I'd be in favor of using a completely different hokusai subcommand for the commit-level output, like `hokusai git-log`. Eventually it could maybe support a `--grep` option to help identify just PRs.

Reply:
oooh interesting diff command in `deploy_pr`. 

I was also thinking of showing all commits and then we can `grep` for only seeing PR commits. With current code it looks like this
```
Changes not staged for commit:
commit 98664694c7f1ac7394c50f6f7d133c82dc100ef2 (upstream/master, master)
Merge: a3968fb 725b658
Author: Chung-Yi Chi <cychi1210@gmail.com>
Date:   Wed Sep 6 13:39:59 2017 -0400

    Merge pull request #266 from ashkan18/post-refund-notification

    Make sure to post notification events when refunded

commit a3968fb08dba34a552fe61a4b291b98075a8e7af
Merge: e293196 1a83704
Author: Ashkan Nasseri <ashkan18@users.noreply.github.com>
Date:   Tue Sep 5 16:12:30 2017 -0400

    Merge pull request #265 from ashkan18/relax-artwork-group-title

    Relax artwork_group.title required flag
```

Now that i've looked at it it actually kind of looks ok, what do you think?

Reply:
I'm wondering if assuming the "Git-flow" PR-style pattern is too much of a generalization... I am in favor of showing the commit details with a flag, then grepp'ing for PRs / merge-commits, if that's what the user wants to do.  I would suggest simplifying and dropping the `--merges` flag from the `git log` command, and changing the option name to `--commits` (the internal flag to `commits`). WDYT?

Reply:
Or, changing the default `hokusai diff` behavior to show only a `git log` list of commits, then adding mutually exclusive options `--merges` which would invoke, as @ashkan18 you have it now `git log --merges --right-only` and `--full`, which would invoke `git diff`